tests/integration/helpers_deployments.py

```diff
@@ -119,46 +119,61 @@ async def get_unit_hostname(ops_test: OpsTest, unit_id: int, app: str) -> str:
     return hostname.strip()
 
 
+async def unit_from_raw(
+    ops_test: OpsTest,
+    unit_name: str,
+    raw_unit: Dict[str, Any],
+    app_name: str,
+    raw_app: Dict[str, Any],
+    subordinate: bool = False,
+) -> Unit:
+    """Create a Unit object from raw unit data."""
+    unit_id = int(unit_name.split("/")[-1])
+
+    app_id = f"{ops_test.model.uuid}/{app_name}"
+    app_short_id = md5(app_id.encode()).hexdigest()[:3]
+    if subordinate:
+        machine_id = -1
+    else:
+        machine_id = int(raw_unit["machine"])
```

Comment:
nit, for conciseness:
```suggestion
    machine_id = -1 if subordinate else int(raw_unit["machine"])
```